NDTensors/src/linearalgebra/linearalgebra.jl

```diff
@@ -388,23 +367,122 @@ function LinearAlgebra.eigen(
   return D, V, spec
 end
 
-function LinearAlgebra.qr(T::DenseTensor{ElT,2,IndsT}; kwargs...) where {ElT,IndsT}
-  positive = get(kwargs, :positive, false)
-  # TODO: just call qr on T directly (make sure
-  # that is fast)
-  if positive
-    QM, RM = qr_positive(matrix(T))
-  else
-    QM, RM = qr(matrix(T))
-  end
+function qr(T::DenseTensor{<:Any,2,IndsT}; positive=false, kwargs...) where {IndsT}
+  qxf = positive ? qr_positive : qr
+  return qx(qxf, T; kwargs...)
+end
+function ql(T::DenseTensor{<:Any,2,IndsT}; positive=false, kwargs...) where {IndsT}
+  qxf = positive ? ql_positive : ql
+  return qx(qxf, T; kwargs...)
+end
+
+#
+#  Generic function for qr and ql decomposition of dense matrix.
+#  The X tensor = R or L.
+#
+function qx(qx::Function, T::DenseTensor{<:Any,2,IndsT}; kwargs...) where {IndsT}
```

Comment:
Can drop `IndsT` and `where {IndsT}` in the function signature and use `IndsT = indstype(T)` inside the function.

Reply:
Done